src/Uno.Toolkit.UI/Controls/Shadows/ShadowCollection.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System.Collections.ObjectModel;
+using System.Linq;
+
+namespace Uno.Toolkit.UI;
+
+public class ShadowCollection : ObservableCollection<Shadow>
+{
+	public bool HasInnerShadow() => this.Any(s => s.IsInner);
+
+	public string ToKey(double width, double height, Windows.UI.Color? contentBackground) 
```

Comment:
If this key is used in any sort of persistence, it may be vulnerable to culture changes (because of width/height formatting). Let's use invariant formatting instead.